src/interactive/documentation.ts

```diff
@@ -0,0 +1,227 @@
+        },
+        err => {
+            vscode.window.showErrorMessage(LS_ERR_MSG)
+            return ''
+        }
+    )
+}
+
+function setDocumentation(inner: string) {
+    if (!inner) { return }
+    forwardStack = [] // initialize forward page stack for manual search
+    showDocumentationPane()
+    const html = createWebviewHTML(inner)
+    _setHTML(html)
+}
+
+function createWebviewHTML(inner: string) {
+    const darkMode: boolean = vscode.workspace.getConfiguration('julia.documentation').darkMode
+
+    const assetsDir = path.join(extensionPath, 'libs', 'documenter')
+    const googleFonts = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, 'google_fonts')))
+    const fontawesome = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, 'fontawesome.min.css')))
+    const solid = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, 'solid.min.css')))
+    const brands = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, 'brands.min.css')))
+    const katex = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, 'katex.min.css')))
+    const require = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, 'require.min.js')))
+    const documenterScript = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, 'documenter.js')))
+    const documenterStylesheet = panel.webview.asWebviewUri(vscode.Uri.file(path.join(assetsDir, darkMode ? 'documenter-dark.css' : 'documenter-light.css')))
+
+    return `
+<!DOCTYPE html>
+<html lang="en" class=${darkMode ? 'theme--documenter-dark' : ''}>
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Julia Documentation Pane</title>
+    <link href=${googleFonts} rel="stylesheet" type="text/css" />
+    <link href=${fontawesome} rel="stylesheet" type="text/css" />
+    <link href=${solid} rel="stylesheet" type="text/css" />
+    <link href=${brands} rel="stylesheet" type="text/css" />
+    <link href=${katex} rel="stylesheet" type="text/css" />
+    <script src=${require} data-main=${documenterScript}></script>
+    <link href=${documenterStylesheet} rel="stylesheet" type="text/css">
+
+    <script type="text/javascript">
+        const vscode = acquireVsCodeApi()
+        window.onload = () => {
+            const els = document.getElementsByTagName('a')
+            for (const el of els) {
+                const href = el.getAttribute('href')
+                if (href.includes('julia-vscode/')) {
+                    const mod = href.split('/').pop()
+                    el.onclick = () => {
+                        vscode.postMessage({
+                            method: 'search',
+                            params: {
+                                word: el.text,
+                                mod
+                            }
+                        })
+                    }
+                }
+            }
+        }
```

Comment:
Can't we do all of this in the extension, rather than in the webview?

Reply:
Unfortunately no, what I was using here is [the message passing API between vscode and webview](https://code.visualstudio.com/api/extension-guides/webview#scripts-and-message-passing), and this code needs to live within webview
(note that we currently don't use this code, since we disable the link handling for now)